tests/conftest.py

```diff
@@ -117,6 +117,30 @@ def chdir(path):
 #
 
 
+@pytest.fixture(autouse=True)
+def temp_cwd(tmpdir):
+    """
+    Fixture to create new temporary directory and use it as working directory.
+    This ensures a clean start for each test and prevents tests from affecting
+    another one through changes to the working directory.
+    """
+    cwd = tmpdir.mkdir('temp-cwd')
+    with chdir(cwd):
+        yield
```

Comment:
No need to create an additional subdirectory, the `tmpdir` will be empty. Also the fixture name could be a bit more descriptive.

```python
def setup_teardown_cwd_tmpdir(tmpdir):
    """
    Fixture to change the working directory to a temporary directory.
    This ensures a clean start for each test and prevents tests from affecting
    another one through changes to the working directory.
    """
    with chdir(tmpdir):
        yield
```